diff --git a/lib/Session.cpp b/lib/Session.cpp
index 3a88264..c0903fd 100644
--- a/lib/Session.cpp
+++ b/lib/Session.cpp
@@ -599,6 +599,13 @@ void Session::close()
 
     if (isRunning())
     {
+#ifdef Q_OS_WIN
+        _shellProcess->closePty();
+        if (!_shellProcess->waitForFinished(1000))
+        {
+            QTimer::singleShot(1, this, SIGNAL(finished()));
+        }
+#else
         // Try SIGHUP, and if unsuccessful, do a hard kill.
         // This is the sequence used by most other terminal emulators like xterm, gnome-terminal, ...
         if (sendSignal(SIGHUP))
@@ -617,6 +624,7 @@ void Session::close()
                 QTimer::singleShot(1, this, SIGNAL(finished()));
             }
         }
+#endif
     }
     else
     {
diff --git a/lib/Vt102Emulation.cpp b/lib/Vt102Emulation.cpp
index 19d9fcc..e6d38e0 100644
--- a/lib/Vt102Emulation.cpp
+++ b/lib/Vt102Emulation.cpp
@@ -26,7 +26,9 @@
 
 // Standard
 #include <cstdio>
+#ifndef Q_OS_WIN
 #include <unistd.h>
+#endif
 
 // Qt
 #include <QEvent>
